packages/realm/src/errors.ts

```diff
@@ -138,3 +140,45 @@ export class ClientResetError extends SyncError {
     };
   }
 }
+
+/**
+ * An error class that indicates that one or more object changes have been reverted by the server.
+ * This can happen when the client creates/updates objects that do not match any subscription, or performs writes on
+ * an object it didn't have permission to access.
+ */
+export class CompensatingWriteError extends SyncError {
+  /**
+   * The array of compensating writes performed by the server.
```

Comment:
Should there be a distinction between?:
* (A) The information about an object that caused the compensating write error.
* (B) A compensating write.

If so, one reformulation could be
```suggestion
   * The array of information about each object that caused the compensating write.
```